File: Strings/reverse_the_string.py
# Given an input string, reverse the string word by word.

class Solution:
    # @param A : string
    # @return string
    def reverseWords(self, A):
        words = A.split()
        ans = ""
        for i in range(len(words)-1,-1,-1):
            if " " not in words[i]:
                ans = ans + words[i] if i == 0 else ans + words[i]+" "
        return ans

File: Stacks and Queues/reverse_the_string.java
// Given a string S, reverse the string using stack.

public class Solution {
    public String reverseString(String A) {
        Stack<Character> s = new Stack<Character>();
        for(int i=0;i<A.length();i++){
            s.push(A.charAt(i));
        }
        String ans = "";
        while(s.isEmpty() == false){
            ans += s.pop();
        }
        return ans;
    }
}
